File: multilibrary/src/main/java/com/zhongjh/albumcamerarecorder/album/listener/OnLoadAllAlbumListener.java
package com.zhongjh.albumcamerarecorder.album.listener;

import com.zhongjh.albumcamerarecorder.album.entity.Album2;

import java.util.List;

/**
 * 返回数据的事件
 *
 * @author zhongjh
 * @date 2022/9/9
 */
public interface OnLoadAllAlbumListener {

    /**
     * 查询完成后会回调该事件
     *
     * @param data 查询后的数据源
     */
    void onLoadAllAlbumComplete(List<Album2> data);

}

File: multilibrary/src/main/java/com/zhongjh/multimedia/album/listener/OnLoadAllAlbumListener.kt
package com.zhongjh.multimedia.album.listener

import com.zhongjh.multimedia.album.entity.Album

/**
 * 返回数据的事件
 *
 * @author zhongjh
 * @date 2022/9/9
 */
interface OnLoadAllAlbumListener {
    /**
     * 查询完成后会回调该事件
     *
     * @param data 查询后的数据源
     */
    fun onLoadAllAlbumComplete(data: List<Album?>?)

    /**
     * 异常信息
     * @param t 异常
     */
    fun onFail(t: Throwable?)
}
